In [1]:
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np

In [2]:
data=pd.read_csv('/content/car_age_price.csv')

In [3]:
data.head()

Year   Price
0  2018  465000
1  2019  755000
2  2019  700000
3  2018  465000
4  2018  465000

In [4]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 112 entries, 0 to 111
Data columns (total 2 columns):
 #   Column  Non-Null Count  Dtype
---  ------  --------------  -----
 0   Year    112 non-null    int64
 1   Price   112 non-null    int64
dtypes: int64(2)
memory usage: 1.9 KB


In [6]:
data.isna().sum()

Year     0
Price    0
dtype: int64

In [51]:
x=data.drop(['Price'],axis=1)
y=data['Price']

In [52]:
from sklearn.model_selection import train_test_split
x_train,x_test,y_train,y_test=train_test_split(x,y,test_size=0.2)

**LINEAR REGRESSION**

In [53]:
from sklearn.linear_model import LinearRegression

In [67]:
from sklearn.metrics import mean_squared_error,r2_score
lrn=LinearRegression()

In [85]:
ln=lrn.fit(x_train,y_train)

In [86]:
y_pred=lrn.predict(x_test)

In [87]:
mean_squared_error(y_test,y_pred)

3090589717.4251018

In [88]:
r2_score(y_test,y_pred)

0.6305430780849538

**LASSO REGRESSION**

In [39]:
from sklearn.linear_model import Lasso

In [41]:
ls=Lasso()

In [81]:
lsm=ls.fit(x_train,y_train)
y_pred1=lsm.predict(x_test)

In [98]:
mean_squared_error(y_test,y_pred1)

3090604029.0483766

In [99]:
r2_score(y_test,y_pred1)

0.6305413672372626

**2022 PRICE**

In [93]:
f_pred=pd.DataFrame({'Year':[2022]})

In [94]:
lr_pred=ln.predict(f_pred)

In [95]:
np.round(lr_pred,2)

array([705479.63])

The predicted price is 705479.63 in 2022

Linear regression is the better model since the mean squared error is lower than lasso regression